src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -158,18 +157,18 @@ static int32_t peep_move_one_tile(Direction direction, Peep* peep)
         if (direction == 0 || direction == 2)
         {
             // Peep is moving along X, so apply the offset to the X position of the destination and clamp their current Y
-            peep->DestinationX += offset;
+            peep->Destination.x += offset;
             const uint16_t centerLine = (peep->y & 0xFFE0) + COORDS_XY_HALF_TILE;
-            peep->DestinationY = std::clamp(
+            peep->Destination.y = std::clamp(
                 peep->y, static_cast<int16_t>(centerLine - 3), static_cast<int16_t>(centerLine + 3));
```

Comment:
I think you can make everything int32_t here and do away with the casting.

Reply:
Some of these places change how the game work if we change them to 32 bits, gotta be careful (I don't remember which ones though, just faced this once when I was migrating to CoordsXY(ZD))